src/components/interface/AudioPlay.jsx

```diff
@@ -1,26 +1,78 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+
 import { withStyles } from '@material-ui/core/styles';
 import PlayCircleOutlineIcon from '@material-ui/icons/PlayCircleOutline';
+
 import styles from '../styles';
+import Button from '../interface/Button';
+import * as firebase from 'firebase';
 
 class AudioPlayer extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      disabled: false
+      audioFile: undefined,
+      metadataLoaded: false
     };
   }
 
-  // TODO: handle audio playing file
+  loadAudio = () => {
+    this.pathRef.getDownloadURL().then((url) => {
+      fetch(url).then(() => {
+        this.setState({ audioFile: new Audio(url) });
+        this.state.audioFile.addEventListener('loadeddata', () => {
+          this.setState({ metadataLoaded: true });
+        })
+      })
+    }).catch(function () {
+      console.log("ERROR. Firebase audio file failed to load.");
+    });;
+  }
+
+  componentDidMount() {
+    const { audioURL } = this.props;
+    const { audioFile } = this.state;
+
+    this.pathRef = firebase.storage().ref().child(audioURL);
+    if (!audioFile) {
+      this.loadAudio();
+    }
+  }
+
+  componentWillUnmount() {
+    const { audioFile } = this.state;
 
+    if (audioFile) {
+      audioFile.pause();
+    }
+  }
+
+  handleClick = () => {
+    const { audioFile } = this.state;
+
+    if (audioFile.paused) {
+      audioFile.play();
+    } else {
+      audioFile.pause();
+    }
+  }
 
-  // TODO: create method to convert time to XX:XX format
+  formattedAudioDuration = () => {
+    const { audioFile } = this.state;
+
+    const secs = audioFile.duration;
+    const formattedTime = Math.floor(secs / 60) + ":" + Math.floor(secs % 60);
+
+    return formattedTime;
+  }
 
   render() {
-    const { fileName, time } = this.props;
-    const { disabled } = this.state;
+    const { audioURL } = this.props;
+    const { audioFile, metadataLoaded } = this.state;
+    const fileName = firebase.storage().ref().child(audioURL).name;
```

Comment:
Inefficient to retrieve the file at `audioURL` every time the component renders. `fileName` has been made into a `state` property and it is assigned when the component mounts.